modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -145,9 +145,10 @@ func (k Keeper) GetAllActiveChannels(ctx sdk.Context) []genesistypes.ActiveChann
 		keySplit := strings.Split(string(iterator.Key()), "/")
 
 		ch := genesistypes.ActiveChannel{
-			ConnectionId: keySplit[2],
-			PortId:       keySplit[1],
-			ChannelId:    string(iterator.Value()),
+			ConnectionId:        keySplit[2],
+			PortId:              keySplit[1],
+			ChannelId:           string(iterator.Value()),
+			IsMiddlewareEnabled: k.IsMiddlewareEnabled(ctx, keySplit[1], string(iterator.Value())),
```

Comment:
In the future, I think it'd probably be wise to set these key splits/iterator values to variables for readability purposes. I don't think it needs to change as this is a problem throughout our codebase (keysplit should probably be a parse func)

Reply:
Agreed! :)